### DISCUSSION TITLE
Back to the Roots

### DISCUSSION THREAD

**Discussion by wartungsmonteur** [wills106/homeassistant-solax-modbus#984] [Q&A]
Hallo

Das Solax Plugin war mal richtig gut und alle Entitäten gingen ohne Probleme.  Auch Updates liefen super. Dann irgendwann kam der Cut und plötzlich waren viele Entitäten nicht mehr erreichbar bzw.  da, aber inaktiv. Mittlerweile mache ich keine Updates mehr und lasse das kastrierte Plugin so wie es ist. Schade das dies so ist.  Bitte geht wieder zum Anfang zurück und holt die alte Verdion aus der Vetsenkung. Danke für eure tolle Arbeit.

**Comment by steps56** [comment]
Sehe ich exakt ebenso.
Daher ging ich schon vor vielen Versionen zurück auf 2024.05.3 und bleibe auch da.
Danach war die Integration kastriert und in vielen Bereichen nicht nutzbar bzw. fehlte Vieles.

Warum zudem die Entität
sensor.solax_fvrt_function
und deren 2 lower/upper Werte zwar auftauchen, aber seltsamerweise nicht wie ähnliche andere dann einstellbar sind, ist mysteriös.

Ich habe in der solax.py mit Einigem dazu experimentiert, habe dazu aber zu wenig Wisden, um es  selbst hin zu biegen.
Woran liegt es, dass dieser Wert mit einstellbar ist (enable/disable) wie in der Qcells/SolaX App ?

Die anfangs top Integration wurde leider seit 2024.05.3 immer weiter verstümmelt und verschlimmbessert.
Daher so nicht nutzbar für mich.
War svhon Aktion dann den nervigen permanenten Updatehinweis mittels Automation abstellen zu können.
Sorry, aber gegenüber früher ist das seither nur in den letzten 8 Versionen nur eine Rückwärtsevolution.

Back to the nice days roots and implenetation of missing/not working stuff like mentioned bevor please.

**Comment by realquackerjack** [comment]
I feel the same way. 
Why had the firmwareinfos (DSP Major/Minor, ARM Majpr/Minor) get to be only visible in the deviceinfo (internal) or how can I access the internal variables/info to work with them in HA?
To patch the entities back in every new release is not what I intended to do every time.

**Comment by HA-Supi** [comment]
+1 from me. 
Please let us access these internal Entities for our own HA Tile. ( sensor.solax_inverter_arm_firmware_minor_version / New sensor.solax_firmware_arm_minor_version) and so on.

**Comment by dbernt55** [comment]
+1 from me!
This integration worked extremely well since Ma 23 when I began to use it. This year however it entered a downslope which seems to continue forcing one either to change code ourselves or to remain on older versions.
@Wills106: what can be done to get it back on track?
Regards
Bernt
